src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2687,6 +2687,10 @@ internal BoundExpression GenerateConversionForAssignment(TypeSymbol targetType,
                         Error(diagnostics, ErrorCode.ERR_NoImplicitConv, syntax, distinguisher.First, distinguisher.Second);
                     }
                 }
+                else if (sourceType.SpecialType == SpecialType.System_Void)
```

Comment:
>else if (sourceType.SpecialType == SpecialType.System_Void) [](start = 16, length = 59)

Would it be better to teach SymbolDistinguisher (or the other component that causes a crash) to properly deal with void type?